spec/p2p/v0.34/README.md

```diff
@@ -0,0 +1,71 @@
+# Peer-to-Peer Communication
+
+This document describes the implementation of the peer-to-peer (p2p)
+communication layer in Tendermint.
+
+It is part of an [effort](https://github.com/tendermint/tendermint/issues/9089)
+to produce a high-level specification of the operation of the p2p layer adopted
+in production Tendermint networks.
+
+This documentation, therefore, considers the releases `0.34.*` of Tendermint, more
+specifically, the branch [`v0.34.x`](https://github.com/tendermint/tendermint/tree/v0.34.x)
+of this repository.
+
+## Overview
+
+A Tendermint network is composed by multiple Tendermint instances, thereafter
+called **nodes**, that interact by exchanging messages.
+
+Tendermint assumes a partially-connected network model.
+This means that a node is not assumed to be directly connected to every other
+node in the network.
+Instead, each node is directly connected to a subset of other nodes in the
+network, thereafter called its **peers**.
+
+The peer-to-peer (p2p) communication layer is responsible for establishing
+connections between nodes in a Tendermint network,
+for managing the communication between a node and its peers,
+and for intermediating the exchange of messages between peers in Tendermint protocols.
+
+## Contents
+
+The documentation follows the organization of the `p2p` package of Tendermint,
+which implements the following abstractions:
+
+- [Transport](./transport.md): establishes secure and authenticated
+   connections with peers;
+- [Switch](./switch.md): responsible for dialing peers and accepting
+   connections from peers, for managing established connections, and for
+   routing messages between the reactors and peers,
+   that is, between local and remote instances of the Tendermint protocols;
+- [PEX Reactor](./pex.md): responsible for peer discovery, ranking, and
+   management, roles that are implemented by:
+   - [Address Book](./addressbook.md): tracks information about other nodes
+      in the network and allows ranking peers based on quality metrics;
+   - [Peer Exchange protocol](./pex-protocol.md): enables nodes to exchange
+      peer addresses, thus implementing a peer discovery mechanism;
+   - [Peer Manager](./peer_manager.md): responsible for establishing
+      connection with peers, defining when and which peers a node should dial;
+- Finally, [Types](./types.md) and [Configuration](./configuration.md) provide
+   a list of existing types and configuration parameters used by the p2p layer implementation.
+
+## Further References 
```

Comment:
We should probably remove those files once this spec is stable, right?

Reply:
Agreed for `types.md`, the other should be improved and merged with the existing list of configuration parameters and their effect.